### PR TITLE
Feature/quiz functionality

### PR DESCRIPTION
**Pull request by tbongiorno1**

### CODE REVIEW

File: client/components/guest/game-components/Question.jsx [hunk 1]
```diff
@@ -1,14 +1,36 @@
...
     return (
       <div id="question">
-        <div id="question-text">the text of the question goes here</div>
         <p> Question </p>
-        <div id="answerOne">answer one</div>
-        <div id="answerTwo">answer two</div>
-        <div id="answerThree">answer three</div>
+        <div id="question-text">the text of the question goes here</div>
+        <ul>
+        <li id="answerOne" onClick={this.handleAnswerClick}>answer one</li>
```
Comment (nferner) [inline client/components/guest/game-components/Question.jsx]: please indent the list items

File: client/components/guest/game-components/QuestionList.jsx [hunk 2]
```diff
@@ -2,11 +2,69 @@ import React, { Component } from 'react';
 import Question from './Question.jsx';
 
 class QuestionList extends Component {
+  constructor(props) {
```
Comment (arianacofone) [inline client/components/guest/game-components/QuestionList.jsx]: What are the props being brought in this component?

File: dist/index.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
 <head>
   <title>Crank-O-Meter</title>
   <link rel="stylesheet" href="./stylesheets/style.css" media="screen" title="no title">
+  <script src="http://code.jquery.com/jquery-2.1.1.min.js"></script>
```
Comment (nferner) [inline dist/index.html]: i believe we can get rid of this jquery

File: dist/stylesheets/style.css [hunk 4]
```diff
@@ -23,26 +23,47 @@ body {
  h1 {
    color: cornflowerblue;
  }
+ @keyframes rotate {
```
Comment (nferner) [inline dist/stylesheets/style.css]: I believe keyframes is unnecessary

File: client/components/guest/game-components/Question.jsx [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
+      this.props.handleSubmission();
+      this.setState(this.clientScore += 1);
+    }
+    console.log(this.clientScore);
+  }
   render() {
     return (
       <div id="question">
         <p> Question </p>
+        <div id="question-text">{this.props.text}</div>
```
Comment (arianacofone) [inline client/components/guest/game-components/Question.jsx:28]: What are props are being brought in? Can we do a proptypes for that puppy?

File: client/components/guest/game-components/Question.jsx [hunk 6]
```diff
@@ -1,10 +1,38 @@
...
+  // answerChoiceOneHandler() {
+  //   // const answerOneRotate = document.querySelector('#crankStick');
+  //   $('#crankStick').css({ transform: 'rotate(120deg)'});
+  // }
+  handleAnswerClick(e) {
+    const one = document.querySelector('#answerOne');
+    let clicked = e.target.id;
+    debugger;
+    if ( clicked === one ) {
+      this.props.handleSubmission();
```
Comment (arianacofone) [inline client/components/guest/game-components/Question.jsx]: Same question for this. What propTypes are being brought in?

File: client/components/guest/game-components/QuestionList.jsx [hunk 7]
```diff
@@ -16,12 +16,38 @@ class QuestionList extends Component {
...
+        text: 'Super Not Cool?',
+        choices: ['1', '2', '3'],
+      });
+      this.setState({ questions });
+      this.questionCount += 1;
+    } else {
+      console.log(this.questionCount);
+    }
+
     console.log(this.questionCount);
```
Comment (nferner) [inline client/components/guest/game-components/QuestionList.jsx]: don't need this console.log

File: client/components/guest/game-components/QuestionList.jsx [hunk 8]
```diff
@@ -2,11 +2,69 @@ import React, { Component } from 'react';
...
+      questions: [
+        {
+          text: 'Cool?',
+          choices: ['1', '2', '3'],
+        },
+      ],
+    };
+    this.createQuestion = this.createQuestion.bind(this);
+  }
+  createQuestion() {
```
Comment (arianacofone) [inline client/components/guest/game-components/QuestionList.jsx:25]: Woah this looks intense. Nice work!

File: client/components/guest/game-components/QuestionList.jsx [hunk 9]
```diff
@@ -2,11 +2,69 @@ import React, { Component } from 'react';
...
+      this.setState({ questions });
+      this.questionCount += 1;
+    } else {
+      console.log(this.questionCount);
+    }
+
+    console.log(this.questionCount);
+  }
   render() {
+    const questionElements = this.state.questions.map((question, idx) => {
```
Comment (arianacofone) [inline client/components/guest/game-components/QuestionList.jsx:56]: Is this rendering a new question below the existing question list?

File: dist/index.html (hunk 3 above)
Comment (arianacofone) [reply dist/index.html]: I second that. I didn't see any above.

File: npm-debug.log [hunk 10]
```diff
@@ -1,27 +0,0 @@
-0 info it worked if it ends with ok
```
Comment (arianacofone) [inline npm-debug.log:1]: So this is the only conflict and since it's the npm-debug.log, go with what's working on your branch.

**Review by arianacofone — COMMENTED** [review]
Just a couple props questions for you. Nice work!

File: client/components/guest/game-components/QuestionList.jsx (hunk 9 above)
Comment (tbongiorno1) [reply client/components/guest/game-components/QuestionList.jsx:56]: yes, it's mapping over and then concatenating then into the new list.

File: client/components/guest/game-components/Question.jsx (hunk 6 above)
Comment (tbongiorno1) [reply client/components/guest/game-components/Question.jsx]: They're being brought in from questionList where handleSubmission is located.

**Comment by nferner** [comment]
SOOOOOOO, slight dilemma on my part. Since teej and I worked on this together I don't think I can really be a plus 1, so I guess this should go to one of the guys for a plus 2 perhaps. 
Also I know why we are doing this now... I think we bit off a bit more than we could chew and we want to push out some of our work, but because this code is not complete, normally I don't think we would be pushing it to the master. For example in a real life scenario we might want to submit something smaller and complete for review before it got merged. Maybe we can talk about how this could be broken up a little more tomorrow. "Tomorrow, tomorrow, I'll love ya tomorrow. It's only a day away."

File: client/components/guest/game-components/QuestionList.jsx (hunk 9 above)
Comment (arianacofone) [reply client/components/guest/game-components/QuestionList.jsx:56]: Wow, that's neat.

File: client/components/guest/game-components/Question.jsx (hunk 5 above)
Comment (tbongiorno1) [reply client/components/guest/game-components/Question.jsx:28]: I'm not sure.

File: dist/index.html (hunk 3 above)
Comment (tbongiorno1) [reply dist/index.html]: Will do.

**Comment by arianacofone** [comment]
Gotcha Gotcha. Why don't we hold off on the merge request for now until we can break it down a bit more and submit a fully functioning part. Are you two cool with that?

For the +1 thing, I actually think you are a great +1 Neal because you are able to catch things I can't (ex - functionality needs based on the structure of the question/game logic). We can definitely ask someone on the teaching team to look it over, but I do think you are a solid +1

File: client/components/guest/game-components/Question.jsx (hunk 6 above)
Comment (arianacofone) [reply client/components/guest/game-components/Question.jsx]: Cool. Let's do a propTypes definition above the component so we remember :)

**Comment by tbongiorno1** [comment]
Changes made, waiting on +1 and +2 for merge

File: client/components/guest/game-components/QuestionList.jsx [hunk 11]
```diff
@@ -9,38 +9,44 @@ class QuestionList extends Component {
...
     this.createQuestion = this.createQuestion.bind(this);
+    this.changeScore = this.changeScore.bind(this);
+  }
+  changeScore(values) {
+    let clicked = values;
+    console.log(clicked);
+    let clientScore = this.state.clientScore;
+    console.log(clientScore);
+    let total = (clientScore + clicked);
+    console.log(total);
```
Comment (nferner) [inline client/components/guest/game-components/QuestionList.jsx]: let's remove console.log

**Pull request merged**